components/de_amp/browser/test/de_amp_util_unittest.cc

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/de_amp/browser/de_amp_util.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace de_amp {
+
+/** Test helpers */
+bool CheckIfAmpDetected(const std::string& body, std::string* canonical_link) {
+  return FindCanonicalLinkIfAMP(body, canonical_link);
+}
+void CheckFindCanonicalLinkResult(const std::string& expected_link,
+                                  const std::string& body,
+                                  const bool expected_detect_amp) {
+  std::string actual_link;
+  const bool actual_detect_amp = CheckIfAmpDetected(body, &actual_link);
+  EXPECT_EQ(expected_detect_amp, actual_detect_amp);
+  if (expected_detect_amp) {
+    EXPECT_EQ(expected_link, actual_link);
+  }
+}
+void CheckCheckCanonicalLinkResult(const std::string& canonical_link,
+                                   const std::string& original,
+                                   const bool expected) {
+  GURL canonical_url(canonical_link), original_url(original);
+  EXPECT_EQ(expected, VerifyCanonicalLink(canonical_url, original_url));
+}
+
+/** De AMP Util Tests */
+TEST(DeAmpUtilUnitTest, DetectAmpWithEmoji) {
+  const std::string body =
+      "<html ⚡>"
+      "<head>"
+      "<link rel=\"canonical\" href=\"abc\"/>"
+      "</head>"
+      "<body></body>"
+      "</html>";
+  CheckFindCanonicalLinkResult("abc", body, true);
```

Comment:
this doesn't seem correct because later on when you call `GURL("abc")` to verify the link you're going to get an invalid url. Shouldn't this be treated as a relative link?